shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2272,11 +2291,62 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   } else if ([method isEqualToString:kUpdateConfigMethod]) {
     [self updateConfig:args];
     result(nil);
+  } else if ([method isEqualToString:kOnInteractiveKeyboardPointerMoveMethod]) {
+    CGFloat pointerY = (CGFloat)[args[@"pointerY"] doubleValue];
+    [self handlePointerMove:pointerY];
+    result(nil);
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (void)handlePointerMove:(CGFloat)pointerY {
+  double screenHeight = [UIScreen mainScreen].bounds.size.height;
```

Comment:
Is the pointerY based on the screen, or the flutter app?

Since it comes from the Framework, I assume it is based on the flutter app? So In scenarios (add to app, or app extension) when the flutter app does not fill the full screen, the calculation below that determines if the pointer goes overtop of the keyboard should be based on the flutter app's height instead of the screen height, correct?

Maybe instead of the screenHeight, we should use the height of the FlutterViewController's view?

Reply:
> We are moving away from using [UIScreen mainScreen]. I don't think we should block this PR because of this, but we will need to follow up to clean up later. FYI @mossmana

@Matt2D While usually I would agree with not blocking the PR, in this particular case, this is a new introduction of [UIScreen mainScreen]. I would prefer that we don't add new instances. You can refer to https://github.com/flutter/flutter/issues/125496 for how we are handling the deprecation in other places.

Reply:
@cyanglaz and I were discussing this the other day, and it's actually a bit tricky here - we would need to refactor [this code](https://github.com/flutter/engine/pull/43690/files#diff-365b13daa5168d36d5824f220bc472a4a90f4685761947f8daf9dc76e7f8d962R480-R497) to make it re-usable. 

I think I can take this part over. Can make a PR today.

Reply:
@cyanglaz in this case, since this is being used to determine the pointer position on the screen, this is relative to the screen rather than the flutter view.

Reply:
FYI this is the PR promised: https://github.com/flutter/engine/pull/44303

After that gets landed, you can do something like this here: 

```
UIScreen *screen = _viewController.flutterScreenIfViewLoaded;
if (!screen) {
  return;
}
CGFloat screenHeight = screen.bounds.size.height;
// ...
```

Reply:
Thank you for the code snippet, I think the new api is much cleaner! Updated the code using the mentioned solution.